### DISCUSSION TITLE
rAudio-64bit-20240615 tested with Raspberry Pi 3A+?

### DISCUSSION THREAD

**Discussion by MulberryPi** [rern/rAudio#1733] [Bug Reports]
Hello everybody, I have recently discovered rAudio and first wish to thank rern and all others involved for all of your time and effort with this software!

I am running "rAudio-64bit-20240615.img.xz" quite happily on Raspberry Pi 3, but cannot get it running stable on Raspberry Pi 3A+ - the web interface is unstable and inoperable, and I notice the green LED for SD card activity on the Pi stays solid green indefinitely, which is not normal.

I have reimaged the SD card, tried another SD card, but the issue remains. I have tested my Pi3A+ with Rasperry Pi OS and it works correctly, so I don't believe it is any hardware issue.
 
The two main difference with the Pi 3A+ vs 3 is that it only has 512MB RAM instead of 1GB, but has slightly faster CPU, would this cause conflicts?

Has rAudio-64bit-20240615.img.xz been confirmed working on the Pi 3A+?

Thank you!
Mike

**Comment by ceejayemm** [comment]
@MulberryPi 

I have been running an RPi3A+ for a long time, several years, with no problems, currently using 20240719 with an Anker SoundCore2 Bluetooth speaker. I generally connect to web radio stations and my NAS (OMV) based music library via wifi with the Audio Buffer (Player / Options) set to 8192.  This setup is in use most days when it plays for several hours a day. I use the same image on other RPi3B+ installations I have around the house, there isn't a specific image for the 3A+, only a 32 bit version for the very early RPi models.

Sounds to me like a dodgy SD card or  partially failed burn of the image to the SD card.  I tend to use either the RPI Imager or Balena Etcher to burn my SD cards.

I hope this helps.

Chris

**Reply by MulberryPi** [reply]
Thanks Chris, I have tried multiple SD cards, SD card reader/writers, and Ballina etcher along with Raspberry Pi Imager and none works. These cards/readers/image file etc all work fine with multiple Raspberry Pi 3 I have, just not the 3A+. 

I can access the web interface by both WLAN ap and ethernet, which indicates it is booting, but trying to do anything causes the web page to go unresponsive and the only way to terminate the system is by pulling the power. 

I'm baffled

**Reply by ceejayemm** [reply]
Could it be a faulty SD card reader/socket ?  I think the RPi 3A+ can be booted from a USB drive, could you flash the latest image to a small USB drive and try booting from that ?  I do this for one of my RPi 3B+ due to the caes its in which I know is different but it might be worth a try?

Chris

**Comment by R101G** [comment]
@MulberryPi 
Could it be a problem with the kernel version? If you can get ssh access, it might be worth trying a pacman update. Bear in mind #1726 if you do.

**Reply by ceejayemm** [reply]
I have not had a problem with kernels myself, I only use the updates provided by @Rern, that's not to say this might be a issue here but its not something I have come across before with this version of the RPI.  I would however suggest updating to the latest version via i20240615 / 20240719.

Chris

**Reply by R101G** [reply]
@ceejayemm  You are probably right. The constant green led implies that it is having a problem reading something. I have seen this happen when there is a problem with initially updating the library.

**Comment by ceejayemm** [comment]
@MulberryPi  @rern 

I think I have figured out what the problem is with the RPi3A+.  The latest images of rAudio  seem to start with the Local Browser (Features / Others / Browser on RPi) enabled.  This makes the 3A very sluggish.  If you can get to the Features page (be patient) and Disable the Browser on RPi  feature you should find that the 3A+ springs back into life.

Chris

**Reply by MulberryPi** [reply]
Thanks for your input all! Unfortunately even with patience the system would not let me do anything meaningful. 

I seem to have gotten around this issue by imaging SD, loading it up via a Pi 3, then once I have disabled wlan ap and put in my specifc  wlan network settings, moving that SD over to the 3A+ and booting up successfully.

It now seems to be running normally and SD activity lights indicate normal operation.

Thanks for the suggestion about local browser! I have disabled this as a precaution as it'll be headless anyway.

**Comment by MulberryPi** [comment]
...spoke to soon, not long after the above, the system ground to a halt and the SD light indicator went solid again for 10 mins and required the power pulled. 

I also noticed that when SSH into the system, that memory usage (free -m) is essentially 98% utilised!

This is a bare bones system with no USB or any other libraries attached, and the local browser turned off and only my WLAN setting applied. 

There's something different with the Pi3A+ and the Pi 3 when it comes to 20240615

Also, is the image file for 20240719 available? or is the update something which needs to be done within the system console?

**Reply by ceejayemm** [reply]
Sorry to hear this.  I found another 3A+ in the back of my workdrawer I had forgotten I had. I flashed a new SD card to i20240615, booted from this and then updated it to 20240719 (no there isn't an image file for this, you need to use the Addons / Update feature to apply this update). I turned off the Local Browser and added my NAS library.  Its been running without a hitch for around 15 hours so far. Memory usage:

```
rAudio-Test:~  #  free -m
               total        used        free      shared  buff/cache   available
Mem:             415         156          35           0         233         258
Swap:              0           0           0
```

![image](https://github.com/user-attachments/assets/9e42340b-3258-42fe-bcfb-09e45b48d97d)


Not sure what else I can suggest.  May be you do have a 'bad' RPi 3A+ ;-)

Chris

**Reply by MulberryPi** [reply]
Thanks Chris! Your two 3A+ working with no issues does seem to indicate my Pi could be a culprit - BUT it works stable and beautifully with latest Raspberry Pi OS! 

Perhaps faulty RAM registers in the Pi which for some reason have not yet been utilised by the Raspberry Pi OS and hence not produced a symptom. 

Bizarre! I will keep digging and see what goes

:)

**Comment by R101G** [comment]
@MulberryPi 
If I have understood what you say, not having any libraries probably makes your installation unique. I wonder if there could be a bug causing streams to be saved to the memory until it's full.
Have you tried adding a library to see if that makes a difference? Or perhaps @ceejayemm  could try an installation with no libraries?

**Reply by MulberryPi** [reply]
Hi R101G - when I say I have no libraries, I mean no USB storage devices attached, no music on the boot SD card, no streams or anything else set up apart from wifi settings (SSID and password) and local browser disabled. The intention is to run a normal USB HDD with music.

I've attached a basic USB flash drive with a folder and some FLAC files inside, then powered the system up - no difference unfortunately.

**Reply by MulberryPi** [reply]
In case anyone is interested, I tried rAudio-64bit-20240818 today, and get the same issue again.

However, the RPi2/BCM2836 version of the above works perfectly and survived my soak testing, so I am happy!

**Reply by MulberryPi** [reply]
![Capture1](https://github.com/user-attachments/assets/3a695a2e-f1ac-41f1-a8ea-5eba484ec2b7)

My Pi is also v1.1 and different memory from the above working 3A+ example from Chris so something to take into consideration also.

**Comment by ha11oga11o** [comment]
Sorry to be nosy, but when you mentioned "multiple SD cards"; Do you mean new multiple cards or cards which you had on table, drawer or parts pile?

Im asking you because i had delusion and i tried 3 different cards which were lying around will solve same issue. Just purchase any cheap card class 10 (its probably not, but...), and ref-flash it.  I have small hint it will work after that. Also, as usual check PSU, because thats kinda world problem No1 with them. 

Please, post back if you solve this issue. Im really curious, because i doubt RPI will have hardware problems just by reading SD, since you see green light constant at time of happening. 

Cheers!

**Reply by MulberryPi** [reply]
I mean multiple microSD cards which I circulate between my GoPro and a small pile on table yes. 

No I have not tried a brand new card yet, but I do need to buy another one soon so will try it and let you know, however I am not sure if the card is the problem, all the cards I tried work perfectly fine running rAudio, and Raspberry Pi OS (in my other Raspberry pi Zero and Pi 3) and in my GoPro so surely that confirms cards are OK? I also tried them with **h2testw** test tool and tests came back with no error.

**Reply by ha11oga11o** [reply]
Yeah, thats really strange. It has issues reading SD. At least thats mine conclusion reading above. I mean, rAudio is really nice optimized and it runs on various systems. Either really SD (magically) dont like your RPI or it goes undervoltage that much so it corrupt data render it unreadable down stream causing freeze/hang/long stale. 

I really wish you have nerves to troubleshoot this further, just for giggles.

**Reply by MulberryPi** [reply]
Unfortunately apart from buying another Raspberry Pi 3A+ I'm not sure what else to try. I tried to cover everything pretty comprehensively for a couple of hours before raising this issue. I don't believe it's power supply rated, all power supplies work with all my other raspberry pi with no issues. It's not the image file either as the same image file works on both my Pi 3 boards.

Since I am running the 32 bit version of rAudio on the Pi3A+ with perfect performance, I am living with that.

**Reply by ha11oga11o** [reply]
Yeah, sometimes juice dont worth a squeeze :)

Cheers
